scripts/deploy/deploy.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+import logging
+import os
+import subprocess
+import xml.etree.ElementTree as ET
+
+PLUGIN_XML = 'resources/META-INF/plugin.xml'
+PLUGIN_ID = 7412
+PLUGIN_JAR = 'dist/intellij-pants-plugin.jar'
+CHANNEL = 'BleedingEdge'
+REPO = 'https://plugins.jetbrains.com/plugin/7412'
+
+logger = logging.getLogger(__name__)
+logging.basicConfig()
+logger.setLevel(logging.INFO)
+
+
+def get_head_sha():
+  cmd = 'git rev-parse HEAD'
+  p = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+  output, _ = p.communicate()
+  if p.returncode != 0:
+    logger.error("{} failed.".format(cmd))
+    exit(1)
+
+  return output.strip()
+
+
+if __name__ == "__main__":
+
+  subprocess.check_output('git checkout {}'.format(PLUGIN_XML), shell=True)
+
+  sha = get_head_sha()
+  logger.info('Append git sha {} to plugin version'.format(sha))
+
+  tree = ET.parse(PLUGIN_XML)
+  root = tree.getroot()
+
+  # Find the `version` tag then append the head sha to it.
+  version = root.find('version')
+  if version is None:
+    logger.error("version tag not found in %s".format(PLUGIN_XML))
```

Comment:
I think you meant to use `not found in {}`. `%s` won't be substituted by format.

You could skip the format, and use `logger.error('.... %s', ...)` instead.